torch/distributed/distributed_c10d.py

```diff
@@ -1596,6 +1611,11 @@ def destroy_process_group(group: Optional[ProcessGroup] = None):
         pg._wait_for_pending_works()
 
     if group is None or group == GroupMember.WORLD:
+        # shutdown all backends in the order of pg names. shutting down in order because
```

Comment:
is there any requirement that a child group is destroyed before the parent group that it was created from?

if we destroy in the reverse order that we create in, we could circumvent any issue like that.  cc @minsii

Reply:
Good thing about PG name is that it is additive, e.g., first PG name is always PG0, second PG1, third PG2..... So we we want reverse order of How PGs are created, we just need to reverse sort the PG names

Reply:
oh? i did not realize that. we do not let users name PG?

Reply:
https://github.com/pytorch/pytorch/blob/dea15c9fdcd0d8cf09612ea2861541d1e743d5e5/torch/distributed/distributed_c10d.py#L3709

Reply:
Changed the order to be the reverse of creation, e.g., PG3->PG2->PG1-PG0